### PR TITLE
fix whitespace on added comment

### LINKED ISSUE #95: comment can introduce trailing whitespace
for instance, select this and `:comment`

```
1

2
```

### PR DESCRIPTION
**Pull request by brynphillips**
This hopefully will fix #95.
Fingers crossed.

### CODE REVIEW

**Comment by asottile** [comment]
it's a little more subtle than that, it should add `# ` normally, but not when there's an empty line

File: babi/file.py [hunk 1]
```diff
@@ -707,7 +707,10 @@ def _comment_remove(self, lineno: int, prefix: str) -> None:
     def _comment_add(self, lineno: int, prefix: str, s_offset: int) -> None:
         line = self.buf[lineno]
 
-        self.buf[lineno] = f'{line[:s_offset]}{prefix}{line[s_offset:]}'
+        if line == '':
+             self.buf[lineno] = f'{line[:s_offset]}{prefix}{line[s_offset:]}'
```
Comment (asottile) [inline babi/file.py]: I would write:

```python
if not line:
     self.buf[lineno] = prefix
else:
    ...
```

**Review by asottile — COMMENTED** [review]
you'll also want to add a test in tests/features/comment_test.py

probably can take one of the other `tmpdir` tests. write `1\n\n2\n` to a file, select it all, `:comment`, and validate that you get `'# 1\n#\n# 2'`

File: babi/file.py (hunk 1 above)
Comment (brynphillips) [reply babi/file.py]: Ah is that why it doesn't pass the 100% coverage?

Comment (asottile) [reply babi/file.py]: yeah, otherwise I'd be asserting the bug under test 😆

Comment (asottile) [reply babi/file.py]: or, at least, the `if not line` isn't the reason -- the reason is there's no test for the change yet!

Comment (brynphillips) [reply babi/file.py]: I see! I wrote a test. Seeing if this will fix the coverage.

**Review by asottile — APPROVED** [review]

**Comment by brynphillips** [comment]
This cat looks very sad. This is not happy cat I expected.

**Pull request merged**